### ISSUE TITLE
[FreeBSD] Error - Pyload not downloading

### ISSUE THREAD

**Issue by bmalum** [pyload/pyload#1350]
Hey Guys, 
i have pyload running in an 10.1 Jail in FreeBSD and have these Bugs/Problems found:

```bash
13.04.2015 10:34:35 INFO      Starting builtin webserver: 0.0.0.0:8000
  File "pyLoadCore.py", line 667, in <module>
    main()
  File "pyLoadCore.py", line 658, in main
    pyload_core.start()
  File "pyLoadCore.py", line 412, in start
    self.log.info(_("Free space: %s") % formatSize(spaceLeft))
  File "/root/pyload/module/utils.py", line 95, in formatSize
    return "%.2f %s" % (size, sizes[steps])
IndexError: list index out of range
```

and i see this in the log:
```bash
10.04.2015 07:00:25	ERROR	Error executing hooks: maximum recursion depth exceeded
```

Is there a way to fix this cause of a dependence problem?

**Comment by durox** [comment]
I had the same issue. Apparently the formatSize crashes if the available diskspace is reported with more than 1000TiB. Here' is my easy fix:

change the file pyload/module/utils.py at line 95 from

```python
def formatSize(size):
    """formats size of bytes"""
    size = int(size)
    steps = 0
    sizes = ["B", "KiB", "MiB", "GiB", "TiB"]
    while size > 1000:
        size /= 1024
        steps += 1
    return "%.2f %s" % (size, sizes[steps])
```
to
```python
def formatSize(size):
    """formats size of bytes"""
    size = int(size)
    steps = 0
    sizes = ["B", "KiB", "MiB", "GiB", "TiB"]
    while (size > 1000) and (steps < 4):
        size /= 1024
        steps += 1
    return "%.2f %s" % (size, sizes[steps])
```
That should solve this issue quick and dirty. The problem lies perhaps in the method to determine the available diskspace. Because on my system it reports 1747TiB, which is false, the true value is about 7TiB. Maybe it has something to do with FreeNAS or ZFS (in my case).

hope I could help,
durox

**Comment by bmalum** [comment]
@durox same here i'm in a FreeBSD ZFS Jail ... Well did not solve the Problem for me - Pyload does still not download and also nothing in the logs. Anyway thank you :-)
Does Pyload download now for you?

**Comment by ohaucke** [comment]
The main problem is that the calculated available freespace is to "big", in 0.4.10 it should be fixed.
As a quick workaround to fix it see: https://github.com/pyload/pyload/pull/1179
